### DISCUSSION TITLE
how to change the equality constraint during simulation

### DISCUSSION THREAD

**Discussion by yinzikang** [google-deepmind/mujoco#380] [Asking for Help]
Hi, I am trying to simulate the robot without a gripper opening a door with python-version Mujoco in the following fig.
![image](https://user-images.githubusercontent.com/34791648/178655489-d6b39cdf-b7a4-4e18-acdc-29501e2b3144.png)
As you can see, the robot does not contact the door now. So first I need to move the robot and make it contact the door handle. 
Then I can add a 'connect' type equality constraint to make the robot and door move together.

So the question is: is it possible to change the constraint during the simulation? Or I must make the robot contact the door when loading the XML file, which is troublesome to calculate and adjust the init qpos of the robot.

Any advice is welcomed

**Comment by yuvaltassa** [comment]
See #229

**Reply by yinzikang** [reply]
Your solution is wonderful and works well in my model.
Thanks a lot.
